src/queries/get-users.ts

```diff
@@ -1,22 +1,45 @@
 import { ContextReturn } from '../apollo-context/context-return';
 import { AppDataSource } from '../data-source';
 import { User } from '../entity/User';
+import { BadRequestError } from '../errors/bad-request.error';
 import { errorMessages } from '../errors/error-messages';
 import { UnauthorizedError } from '../errors/unauthorized.error';
+import { UsersPage } from './users-page-type';
 
-export async function getUsers(parent: any, args: { limit?: number }, context: ContextReturn) {
+export async function getUsers(parent: any, args: { limit?: number; index?: number }, context: ContextReturn) {
   if (!context.userId) {
     throw new UnauthorizedError(errorMessages.notAuthenticated);
   }
 
   const defaultLimit = 5;
+  const limit = args.limit ? args.limit : defaultLimit;
+  const index = args.index ? args.index : 1;
```

Comment:
```suggestion
  const index = args.index ?? 0;
```

Pela nomenclatura "index" (ou o "offset" que eu sugeri), eu imaginaria que o primeiro (e o padrão) seria 0, não 1 🤔 


Não faz mais sentido ser 0?

Reply:
Então, pegando gancho no comentário que fiz acima, do ponto de vista do front, o index/offset/page começaria com 1, a minha dúvida é: A equipe de front enxerga o index como começando com 0? Porém mostraria começando do 1 para o usuário final, certo?

Eu que trabalhei pouco com front pensaria em 1, mandaria 0 se eu souber que o back faz com 0. 
Como vocês trabalham nesse caso? 
Resumindo meu pensamento: 
-Acredito que o front mostre 1 para o usuário, as paginações começam com 1, logo acredito que o index/offset deva ser 1.
-Porém, do ponto de vista de programação, o index começa com 0, então o front deve saber que começa com 0 e fazer alterações para mostrar para os usuários que começa com 1

Qual das duas visões, a TaqTile trabalha? Ou nenhuma das duas?

Reply:
Vc acha que `offset` começaria no 1 mesmo? `Offset` no 1 não daria a impressão que vc está pulando um usuário?

Existe a nomenclatura que o próprio TypeORM usa tbm, que é o `skip`, essa me parece bem claro que se vc usar 1 ele vai pular 1 usuário.

O "page" é mais ambíguo, me parece que começar no 1 ou no 0 seria questão de convenção, e o client precisaria saber de antemão. Por isso sugeri `offset`, porque fica claro que começa no 0.

Reply:
Entendi, vou considerar a parte técnica mesmo nas próximas nomenclaturas, em relação aos pontos de vista, vou explicar em uma Daily quando tiver a oportunidade.